tests/test_generator.py

```diff
@@ -312,159 +355,160 @@ def test_write_tagged_file(fileutils, generator_instance: Generator):
+    yield [pathlib.Path(path) for path in sorted(glob(paths + "/*"))]
+
+
+@pytest.fixture
+def context(test_dir: str):
+    yield GenerationContext(pathlib.Path(test_dir))
+
+
+@pytest.fixture
+def operations_list(paths_to_copy, paths_to_symlink, paths_to_configure):
+    op_list = []
+    for file in paths_to_copy:
+        op_list.append(CopyOperation(src=file))
+    for file in paths_to_symlink:
+        op_list.append(SymlinkOperation(src=file))
+    for file in paths_to_configure:
+        op_list.append(SymlinkOperation(src=file))
+    return op_list
+
+
+@pytest.fixture
+def formatted_command_list(operations_list: list, context: GenerationContext):
+    new_list = CommandList()
+    for file in operations_list:
+        new_list.append(file.format(context))
+    return new_list
+
+
+def test_execute_commands(
+    operations_list: list, formatted_command_list, generator_instance: Generator
 ):
-    monkeypatch.setattr(
-        "smartsim._core.dispatch._LauncherAdapter.start",
-        lambda launch, exe, job_execution_path, env, out, err: random_id(),
-    )
-    param_set = {"PARAM0": [0, 1], "PARAM1": [2, 3]}
-    tagged_files = sorted(glob(get_gen_configure_dir + "/*"))
-    ensemble = Ensemble(
-        "ensemble-name",
-        "echo",
-        replicas=1,
-        files=EntityFiles(tagged=tagged_files),
-        file_parameters=param_set,
-    )
-    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
-    job_list = ensemble.build_jobs(launch_settings)
-    exp = Experiment(name="exp_name", exp_path=test_dir)
-    _ = exp.start(*job_list)
-    run_dir = listdir(test_dir)
-    jobs_dir = pathlib.Path(test_dir) / run_dir[0] / "jobs"
-
-    def _check_generated(param_0, param_1, dir):
-        assert dir.is_dir()
-        tagged_0 = dir / "tagged_0.sh"
-        tagged_1 = dir / "tagged_1.sh"
-        assert tagged_0.is_file()
-        assert tagged_1.is_file()
-
-        with open(tagged_0) as f:
-            line = f.readline()
-            assert line.strip() == f'echo "Hello with parameter 0 = {param_0}"'
-
-        with open(tagged_1) as f:
-            line = f.readline()
-            assert line.strip() == f'echo "Hello with parameter 1 = {param_1}"'
-
-    _check_generated(0, 3, jobs_dir / "ensemble-name-1-1" / Generator.run_directory)
-    _check_generated(1, 2, jobs_dir / "ensemble-name-2-2" / Generator.run_directory)
-    _check_generated(1, 3, jobs_dir / "ensemble-name-3-3" / Generator.run_directory)
-    _check_generated(0, 2, jobs_dir / "ensemble-name-0-0" / Generator.run_directory)
-    ids.clear()
+    """Test Generator._execute_commands calls with appropriate type and num times"""
+    with (
+        unittest.mock.patch(
+            "smartsim._core.generation.generator.subprocess.run"
+        ) as mock_run,
+    ):
+        generator_instance._execute_commands(formatted_command_list)
+        assert mock_run.call_count == len(operations_list)
```

Comment:
Should this be the length of the `formatted_command_list`?
```suggestion
        generator_instance._execute_commands(formatted_command_list)
        assert mock_run.call_count == len(formatted_command_list)
```

Reply:
yes but also should be same length tho